### PR TITLE
#5854 workaround to avoid populating Second Level Cache from DQL queries with multiple nested DQL aliases

### PR DESCRIPTION
**Pull request by Ocramius**
Provides workaround for #5854

Doesn't really fix the issue though.

The problem is following:

``` sql
SELECT
    a, b, c
FROM
    A a
JOIN
    a.b b
JOIN
    b.c c
```

Caching a query like this one via SLC will have the `DefaultQueryCache` loop over associations of `A`, but using metadata of all DQL aliases/associations.

This workaround prevents that, but also disables caching of fetched `c` instances from this example.

Note: no tests provided, as existing tests already show the regression.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ php:
   - 5.5
   - 5.6
   - 7.0
+  - 7.1
```
Comment (Ocramius) [inline .travis.yml]: Hmm, not yet available. I guess `nightly` would be an alternative

Comment (lcobucci) [inline .travis.yml]: Indeed

File: lib/Doctrine/ORM/Cache/DefaultQueryCache.php [hunk 2]
```diff
@@ -261,6 +261,14 @@ public function put(QueryCacheKey $key, ResultSetMapping $rsm, $result, array $h
             // @TODO - move to cache hydration components
             foreach ($rsm->relationMap as $alias => $name) {
                 $metadata = $this->em->getClassMetadata($rsm->aliasMap[$rsm->parentAliasMap[$alias]]);
+                $className = $metadata->getName();
```
Comment (lcobucci) [inline lib/Doctrine/ORM/Cache/DefaultQueryCache.php]: Shouldn't this change impact on https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Cache/DefaultQueryCacheTest.php#L138?

Comment (Ocramius) [inline lib/Doctrine/ORM/Cache/DefaultQueryCache.php]: I think the design of the test is broken. The current assertions pass, but further details are indeed skipped

**Comment by lcobucci** [comment]
I think that we also have a small thing on https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Cache/DefaultQueryCacheTest.php#L134. Shouldn't it be `$state->getName()` instead of `$city->getName()`?

**Comment by Ocramius** [comment]
Yep, that needs fixing too.

**Comment by lcobucci** [comment]
I believe that we can fix those small problems in order to prevent that `Notice` and then think on a permanent solution.

**Comment by FabioBatSilva** [comment]
@Ocramius I made a couple changes here, hopefully will fix the issue.

**Comment by Ocramius** [comment]
Ok, seems like I'll have to do some `git bisect` on `master`: loads of issues popping up lately...

This patch is good to go meanwhile, thanks @FabioBatSilva!

**Pull request merged**